### PR TITLE
Analog write primitive

### PR DESCRIPTION
**Pull request by tolauwae**
Add the `analog_write` primitive.

### CODE REVIEW

File: src/Primitives/arduino.cpp [hunk 1]
```diff
@@ -608,6 +616,8 @@ def_prim(subscribe_interrupt, threeToNoneU32) {
     uint8_t tidx = arg1.uint32;  // Table Idx pointing to Callback function
     uint8_t mode = arg0.uint32;
 
+    printf("subscribe_interrupt(%i, %i, %i)\n", pin, tidx, mode);
```
Comment (carllocos) [inline src/Primitives/arduino.cpp:619]: can this be removed?

File: src/Primitives/emulated.cpp [hunk 2]
```diff
@@ -454,7 +460,7 @@ def_prim(subscribe_interrupt, threeToNoneU32) {
 }
 
 // Temporary Primitives needed for analogWrite in ESP32
-def_prim(chip_analog_write, threeToNoneU32) {
+def_prim(chip_ledc_set_duty, threeToNoneU32) {
```
Comment (carllocos) [inline src/Primitives/emulated.cpp:463]: This primitive is no longer needed

**Review by carllocos — CHANGES_REQUESTED** [review]
Out of curiosity is the `analogWrite` a newly supported function for the ESP32?

The reason why I ask is because I recall that such a function was not supported for the ESP32 which is also the reason why I wrote the previous primitive `chip_analog_write` (now renamed to `chip_ledc_set_duty`). Now seeing you use the Arduino `analogWrite` makes me doubt whether I overlooked its existence.

File: src/Primitives/arduino.cpp (hunk 1 above)
Comment (tolauwae) [reply src/Primitives/arduino.cpp:619]: Yes. I'll change it to a debug log.

**Comment by tolauwae** [comment]
> Out of curiosity is the analogWrite a newly supported function for the ESP32?
I thought so. I'll double check.

If not, that does pose a conundrum. Do all the Arduino primitives need to be supported for the ESP? Ideally, I'd say yes, but the counter argument is that real Arduino devices may want to use this primitives.

That is an annoying decision we will continue to encounter as long as we haven't added a modular system.
Don't know on what criteria to decided in those cases.

**Comment by tolauwae** [comment]
But compilation works this time around. All CI checks are green.

I'll double check if it works for the fade example tomorrow.

**Comment by tolauwae** [comment]
@carllocos the primitive works fine on ESP32. I actually used it in code for the MobiSys demo.

File: src/Primitives/emulated.cpp (hunk 2 above)
Comment (tolauwae) [reply src/Primitives/emulated.cpp:463]: Maybe developers want both of them? They are slightly different.

Comment (carllocos) [reply src/Primitives/emulated.cpp:463]: indeed good idea

**Review by carllocos — APPROVED** [review]

**Pull request merged**